bot_test.go

```diff
@@ -304,3 +307,406 @@ func TestDeleteBotPermanently(t *testing.T) {
 		require.Equal(t, appErr, err)
 	})
 }
+
+func TestEnsureBot(t *testing.T) {
```

Comment:
Are these tests a copy of the server `EnsureBot` tests?

Reply:
The semantics are all from the server ones, yes, though rewritten for the client and refactored slightly for clarity.